src/Rules.py

```diff
@@ -324,74 +324,35 @@ def allRegionsAccessible(state):
     # Victory requirement
     multiworld.completion_condition[player] = lambda state: state.has("__Victory__", player)
 
-    def convert_req_function_args(func, args: list[str], areaName: str, warn: bool = False):
+    def convert_req_function_args(func, args: list[str], areaName: str):
         parameters = inspect.signature(func).parameters
         knownArguments = ["world", "multiworld", "state", "player"]
         index = 0
         for parameter, info in parameters.items():
             if parameter in knownArguments:
                 continue
 
-            argType = info.annotation
-            optional = False
-            try:
-                if issubclass(argType, inspect._empty): #if not set then it wont get converted but still be checked for valid data at index
-                    argType = str
+            target_type = info.annotation
```

Comment:
Does this support function args that don't have a type annotation?

Reply:
That's a good question. I'll check ASAP. If it's broken here, then it's probably also broken in stable.
The function in helper.py is fine since it's asking for a type, and Python asks for a second arg, idk what gets sent when no type is provided. I'll know later today anyway

Reply:
Yep, no rush. :smiley: Actually just saw the removed line of `if issubclass(argType, inspect._empty):`, which seems like it was probably handling that. Inspect return annotations also do inspect._empty if there isn't one.

Reply:
You were absolutly right! On top of some broken check for type that would catch any simple types instead of just str I didnt include any check for empty. its there now